tests/pytests/unit/utils/scheduler/test_schedule.py

```diff
@@ -0,0 +1,508 @@
+        "run": True,
+        "name": "testjob",
+        "seconds": 60,
+        "_splay": None,
+        "_seconds": 60,
+        "jid_include": True,
+        "maxrunning": 1,
+        "_next_fire_time": datetime.datetime(2018, 11, 21, 14, 8, 53, 903438),
+    }
+
+    with patch.object(salt.utils.schedule, "log") as log_mock:
+        with patch("salt.utils.process.daemonize"), patch("sys.platform", "linux2"):
+            schedule.handle_func(False, "test.ping", data)
+            assert log_mock.exception.called
+
+
+def test_handle_func_check_data(setup_teardown_vars):
+    """
+    Tests handle_func to ensure that __pub_fun_args is not
+    being duplicated in the value of kwargs in data.
+    """
+    schedule = setup_teardown_vars["schedule"]
+
+    data = {
+        "function": "test.arg",
+        "_next_scheduled_fire_time": datetime.datetime(2018, 11, 21, 14, 9, 53, 903438),
+        "run": True,
+        "args": ["arg1", "arg2"],
+        "kwargs": {"key1": "value1", "key2": "value2"},
+        "name": "testjob",
+        "seconds": 60,
+        "_splay": None,
+        "_seconds": 60,
+        "jid_include": True,
+        "maxrunning": 1,
+        "_next_fire_time": datetime.datetime(2018, 11, 21, 14, 8, 53, 903438),
+    }
+
+    with patch("salt.utils.process.daemonize"), patch("sys.platform", "linux2"):
+        with patch.object(schedule, "standalone", return_value=True):
+            # run handle_func once
+            schedule.handle_func(False, "test.arg", data)
+
+            # run handle_func and ensure __pub_fun_args
+            # is not in kwargs
+            schedule.handle_func(False, "test.arg", data)
+
+            assert "kwargs" in data
+            assert "__pub_fun_args" not in data["kwargs"]
+
+
+def test_handle_func_check_dicts(setup_teardown_vars):
+    """
+    Tests that utils, functions, and returners dicts are not
+    empty after handle_func has run on Windows.
+    """
+    schedule = setup_teardown_vars["schedule"]
+
+    data = {
+        "function": "test.arg",
+        "_next_scheduled_fire_time": datetime.datetime(2018, 11, 21, 14, 9, 53, 903438),
+        "run": True,
+        "args": ["arg1", "arg2"],
+        "kwargs": {"key1": "value1", "key2": "value2"},
+        "name": "testjob",
+        "seconds": 60,
+        "_splay": None,
+        "_seconds": 60,
+        "jid_include": True,
+        "maxrunning": 1,
+        "_next_fire_time": datetime.datetime(2018, 11, 21, 14, 8, 53, 903438),
+    }
+
+    with patch("salt.utils.process.daemonize"):
+        with patch.object(schedule, "standalone", return_value=True):
+            # simulate what happens before handle_func is called on Windows
+            schedule.functions = {}
+            schedule.returners = {}
+            schedule.utils = {}
+            schedule.handle_func(False, "test.arg", data)
+
+            assert {} != schedule.functions
+            assert {} != schedule.returners
+            assert {} != schedule.utils
```

Comment:
```suggestion
            assert schedule.functions != {}
            assert schedule.returners != {}
            assert schedule.utils != {}
```
assert what you get against what you expect.
In case of errors, the diff by pytest will make more sense.